updated triage state

s-santillan opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

- [ ] Update with screenshots

s-santillan commented:
There's only one mention of Closed so it should be straightforward. I will have to update the screenshots (so the docs will follow after the push to prod. FYI @chmccreery)

chmccreery commented on docs/semgrep-app/findings.md:
```diff
@@ -69,7 +69,7 @@ Semgrep App assists in the triage process through the use of **comments** and **
 | **Open** | Open findings require action, such as rewriting the code for vulnerabilities, or refactoring the code. Findings are open by default. |
 | **Ignored** | Findings that are ignored will not be acted upon. This may be a false positive or deprioritized issue. Findings can be ignored through Semgrep App (see [Managing triage states](#managing-triage-states-bulk-triage)). |
-| **Closed** | Findings are considered closed under any of the following conditions: <ul><li>After a new scan, the code has been addressed such that the rule pattern no longer matches the code.</li><li>The rule has been removed from your Rule Board.</li></ul> |
+| **Fixed** | Fixed findings are findings that were detected in a previous scan but no longer trigger a match in the most recent scan. The rule that detected the finding and the code that triggered the match must both be active in the previous and most recent scan. |
```
```suggestion
| **Fixed** | Fixed findings are findings that were detected in a previous scan but no longer trigger a match in the most recent scan. The rule that detected the finding and the code that triggered the match must both be active in the most recent scan. |
```

chmccreery reviewed (approved):
Looks good! Can you also update the screenshot in https://semgrep.dev/docs/semgrep-app/findings/#managing-triage-states-bulk-triage?

chmccreery commented:
Actually maybe it's worth adding a section for "removed" here and calling out that removed findings are not shown in the UI but there are two ways to remove a finding:
- remove the rule that triggered it from your rule board and re-scan
- add the file to your .semgrepignore

What do you think? This way users who want to get rid of their old findings have a path forward and users who were used to our previous behavior know why their removed findings disappeared

chmccreery commented:
Can we hold off on merging this, though? @raghavjain3 is doing a final check with customers before we merge the semgrep-app changes to make sure nobody really cared about seeing their removed findings

s-santillan commented:
> Looks good! Can you also update the screenshot in https://semgrep.dev/docs/semgrep-app/findings/#managing-triage-states-bulk-triage?

Yep, I'll be updating all the screenshots if/when this goes out to prod. Won't be merging this PR until then.

> Actually maybe it's worth adding a section for "removed" here and calling out that removed findings are not shown in the UI but there are two ways to remove a finding:
> 
>     * remove the rule that triggered it from your rule board and re-scan
> 
>     * add the file to your .semgrepignore
> 
> 
> What do you think? This way users who want to get rid of their old findings have a path forward and users who were used to our previous behavior know why their removed findings disappeared

There's a "how to remove the rule that triggered the finding". I'll add the suggestion to add the file to .semgrepignore :) Thank you.

chmccreery commented on docs/semgrep-app/findings.md:
```diff
@@ -124,9 +122,10 @@ To **open findings**:
 5. Optional: Include a **comment** explaining the action.
 
-To **close findings**, do one of the two options:
+To **fix a finding**, update, or refactor the code such that the Semgrep rule pattern no longer matches the code.
```
This is great! Can we also add something like:
```suggestion
To **fix a finding**, update, or refactor the code such that the Semgrep rule pattern no longer matches the code.

There are two ways to **remove a finding** altogether: either remove the rule that found it or remove the file it is in from the list of files scanned by semgrep. Then, re-scan the branch or pull request containing the issue. The issue should then be removed.
```

s-santillan replied:
(Have added this in!)

s-santillan commented:
Hello @chmccreery and @adamkvitek!

I would like to request a review of this document as I rewrote about half of it, specifically the introduction part, to explain "what do we do with findings and what makes the management of findings unique to app (i.e. mass triage and deduplication). It may be best to view the file in full (not as a diff). I have added a changelog here because the PR is much expanded from its original content.

Changelog:

* Rewrote much of the intro content to include "what do we do with findings and why do we do these actions?"
    * True positive - fix
    * False positive - ignore finding, rewrite rule, or remove code/rule 
    * False negative - added a tip for people to use "semgrep shouldafound" to help with rule writing
* Added a short section on "deduplication", mentioning that findings are generated "per branch" (Clara, I avoided using "per ref" because "per ref" in a broader sense can refer to any pointer. Let me know if this is okay?)
* Explicitly mention ignored via code or app for additional nuance between the difference of "not scanning a file at all" (.semgrepignore) vs "this finding was found and ignored through nosemgrep"
* My goal was to make this the source of truth for findings in Semgrep App, so I avoided linking to the CLI-only "Findings" doc.

Note: There's no change in Sorting, Bulk triage sections, so that should be good.

NOT YET DONE: I need to update with better screenshots. But feel free to review the text!

Thank you!

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -124,31 +149,43 @@ To **open findings**:
 ## Creating Jira tickets from findings
 
-Semgrep supports the creation of Jira tickets from a finding. This feature is available to Team/Enterprise Tier users.
+Semgrep supports the creation of Jira tickets from a finding. This enables developers and project managers to relevant issues within their own project or bug tracking environment. This feature is available to Team/Enterprise Tier users.
```
I totally missed that. Thank you!

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -124,31 +149,43 @@ To **open findings**:
+
+<dl>
+    <dt>Full scan</dt>
+    <dd>Scans the repository in its entirety. Full scans are performed on mainline branches, such as `master` or `main`. This scan is performed on a scheduled basis.</dd>
```
Hi.  We'll probably go with "It is recommended to perform full scans only on mainline branches." (Not being pedantic, I hope - it's based on style guidelines, see below.) 

Based on my understanding, users can still run a full scan on a non-mainline branch especially now that we've given the option to run semgrep login and the near-future improvements on branch awareness from CLI, so only running scans on mainline branches is a recommended workflow.

(Regarding usage of "should" see: [Google style guide - "should"](https://developers.google.com/style/word-list#should).)

s-santillan replied:
(Added this in.)

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+After a finding is generated, developers can:
+
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
```
```suggestion
* **View the Semgrep rule and the matching code**. For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
```

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
+* **Triage the finding.** If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include commenting and ignoring. This is one method to reduce **false positives**.
```
```suggestion
* **Triage the finding**. If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include commenting and ignoring. This is one method to reduce **false positives**.
```

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+Semgrep rules provide a metadata schema to identify common categories such as the above. Semgrep findings include a `message` field that describes the security issue or bug that must be resolved. Findings may also provide an `autofix` field that fixes the issue by creating a suggestion within your source code management (SCM) tool, such as GitHub or GitLab.
 
-After a finding is generated, developers may:
+Another way of categorizing findings is **based on the validity of of the match**:
```
```suggestion
Another way of categorizing findings is **based on the validity of the match**:
```

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -46,41 +65,47 @@ The Semgrep App Finding page displays findings across all projects connected to
 1. Sign in to Semgrep App.
```
Commenting on line above (related issue to why I comment here: https://github.com/isaacs/github/issues/1655)

Blocking: Change `## Viewing data in the Findings page` to `## Viewing data on the Findings page`

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -46,41 +65,47 @@ The Semgrep App Finding page displays findings across all projects connected to
+| **Ignored** | Findings that are ignored will not be acted upon. This may be a false positive or deprioritized issue. Findings can be ignored through Semgrep App (see [Managing triage states](#managing-triage-states-bulk-triage)). There are two types of ignored findings: <dl><dt>Ignored - App</dt><dd>The ignore status was set within Semgrep App.</dd><dt>Ignored - Code</dt><dd>The ignore status w ...
+| **Fixed** | Fixed findings are findings that were detected in a previous scan but no longer trigger a match in the most recent scan. The rule that detected the finding and the code that triggered the match must both be active in the most recent scan. |
+
+Findings can also be **removed**. A removed finding does not count towards a fix rate or total number of findings. A finding is considered removed if it is not found in the most recent scan due to either of the following conditions:
```
```suggestion
Findings can also be **removed**. A removed finding does not count towards a fix rate or the total number of findings. A finding is considered removed if it is not found in the most recent scan due to either of the following conditions:
```

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -124,31 +149,43 @@ To **open findings**:
 4. Click **Open**.
 5. Optional: Include a **comment** explaining the action.
 
-To **close findings**, do one of the two options:
```
Blocking: 

Reffering to these lines:
<img width="1129" alt="Screenshot 2022-06-20 at 13 41 50" src="https://user-images.githubusercontent.com/26669307/174594241-3917fca1-82a3-446a-8535-9deca3a25e4d.png">

Change both occurences of `one-by-one` to `one by one`.

Reference: https://dictionary.cambridge.org/dictionary/english/one-by-one

s-santillan replied:
(Added!)

adamkvitek reviewed (commented):
@s-santillan This is a big work... a lot of things now have much more context when reading this overview document. Maybe later we can add some use case procedural guide elsewhere (in another task and PR - if users will struggle with some things) and figure out examples for users to go through and try things out. Let me know when I can check again. Thank you! 🙇

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+* business or logic bugs
+* matches based on your own custom rules (such as organization-specific authentication logic)
 
+Semgrep rules provide a metadata schema to identify common categories such as the above. Semgrep findings include a `message` field that describes the security issue or bug that must be resolved. Findings may also provide an `autofix` field that fixes the issue by creating a suggestion within your source code management (SCM) tool, such as GitHub or GitLab.
```
(Fixed!)

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+    <dt>False positive</dt>
+    <dd>A false positive is a mismatch between the intended purpose of the rule and the code it matched. A finding is generated but does not meet the rule's intended need. Rules with a high false positivity rate are said to be <strong>noisy</strong>.</dd>
+    <dt>False negative</dt>
+    <dd>A false negative is a finding that should have been found by a rule, but was not. This could be due to a flaw in the rule's logic. See <a href="../release-notes/#additions-1"><code>semgrep shouldafound</code></a>.</dd>
```
@adamkvitek FYI - shouldafound is a specific type of issue, but there can be more :) 

@chmccreery Regarding bugs in semgrep, I think we should show them the github page for issues ([https://github.com/returntocorp/semgrep/issues](https://github.com/returntocorp/semgrep/issues)) because it may be a known issue. Does this work for you?

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
+* **Triage the finding.** If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include commenting and ignoring. This is one method to reduce **false positives**.
```
(Added this in, thank you!)

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
-* custom pattern matches based on your own custom rules
+* security vulnerabilities (such as dangerous function usage)
+* business or logic bugs
+* matches based on your own custom rules (such as organization-specific authentication logic)
```
This one is just a noun ("Matches") with a very long noun modifier phrase ("based on your own custom rules")

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+
+:::tip
+* An additional method to address false positives is to improve the rule. Create [test cases](../writing-rules/testing-rules) to ensure that the rule performs as intended.
+* If a rule from Semgrep Registry is useful but also captures too many false positives, you can reach out to support@semgrep.dev. This will assist rule-writing efforts and improve the quality of rules that you run. 
```
@adamkvitek This is referring to our developer sentiment feature :)

adamkvitek replied:
(Feel free to ignore.) Maybe we can link some developer sentiment doc here. :-)

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -124,31 +149,43 @@ To **open findings**:
-2. Click on the ruleset that contains the rule.
-3. Drag the rule tile to the **Discard rule** section.
+2. Click the ruleset that contains the rule.
+3. Click the garbage can icon next to the rule you're deleting.
```
Added.

s-santillan commented:
@adamkvitek

s-santillan commented:
I agree that the focus is off as there are too many possible workflows in the doc because there are many reasons why a user might be on the page.
I think we're hitting the point where we do need more user story based flows and not mix too many goals in a doc. In this doc alone I:
(1) Tried to list all the features in the page
(2) Offered too many workflows (because they're all doable on that page) and tried to contextualize why a user might want to do any of those
(3) Had some piecemeal procedures.
It should probably be broken up into more docs pages, but how? I feel like I do have a good idea of the journey (at least in App) but also feel weighed down by existing structures in the docs (think of how we'd manage redirects).
So we do want to go from V1 sitemap (docs list features) to V2 sitemap (docs list tasks/user journeys). We originally intended to have user interviews for this transition. However, V2's story-based sitemap feels like a moving target alongside many other moving targets as our product is hitting a high-growth stage (puberty?)
Thoughts are welcome!

s-santillan commented:
Hey @adamkvitek!

Thanks for the feedback. I agree that the focus is off as there are too many possible workflows in the doc because there are many reasons why a user might be on the page.

I think we're hitting the point where we do need more user story based flows and not mix too many goals in a doc. In this doc alone I:
(1) Tried to list all the features in the page
(2) Offered too many workflows (because they're all doable on that page) and tried to contextualize why a user might want to do any of those
(3) Had some piecemeal procedures.

It should probably be broken up into more docs pages, but how? I feel like I do have a good idea of the journey (at least in App) but also feel weighed down by existing structures in the docs (think of how we'd manage redirects).

So we do want to go from V1 sitemap (docs list features) to V2 sitemap (docs list tasks/user journeys). We originally intended to have user interviews for this transition. However, V2's story-based sitemap feels like a moving target alongside many other moving targets as semgrep is hitting a high-growth stage (puberty?)

We should probably check in with the rest of the team about this. I will copy paste this comment where we can loop in the others. Thanks!

adamkvitek commented:
@s-santillan thank you! Replied in our Slack thread. :-)

s-santillan commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
+* **Triage the finding.** If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include commenting and ignoring. This is one method to reduce **false positives**.
+* **Remove the rule or code that generated the finding**. There are cases where Semgrep scans a file not meant for scanning or when a rule is irrelevant. You can remove the rule from the Rule board or add the file to the ignore list.
```
@chmccreery I will add this step in! But AFAIK I don't think there's an easy way from app to trigger a scan at will.

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,44 @@ import MoreHelp from "/src/components/MoreHelp"
+* business or logic bugs
+* matches based on your own custom rules (such as organization-specific authentication logic)
 
+Semgrep rules provide a metadata schema to identify common categories such as the above. Semgrep findings include a `message` field that describes the security issue or bug that must be resolved. Findings may also provide an `autofix` field that fixes the issue by creating a suggestion within your source code management (SCM) tool, such as GitHub or GitLab.
```
Optional, feel free to ignore.

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,49 @@ import MoreHelp from "/src/components/MoreHelp"
+* Business or logic bugs
+* Matches based on your own custom rules (such as organization-specific authentication logic)
 
+Semgrep rules provide a metadata schema to identify common categories such as the above. Semgrep findings include a `message` field that describes the security issue or bug that must be resolved. Findings may also provide a `fix` field that fixes the issue by creating a suggestion within your source code management (SCM) tool, such as GitHub or GitLab.
```
```suggestion
Semgrep rules provide a metadata schema to identify common categories such as the above. Semgrep findings include a `message` field that describes the security issue or bug found in matching code. Findings may also provide a `fix` field that fixes the issue by creating a suggestion within your source code management (SCM) tool, such as GitHub or GitLab.
```

? (just presuming that the issue is not necessarily a “must solve”)

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,49 @@ import MoreHelp from "/src/components/MoreHelp"
+    <dd>A false negative is a finding that should have been found by a rule, but was not. This can happen for two reasons:
+    <ul>
+        <li>A flaw in the rule's logic. See <a href="../release-notes/#additions-1"><code>semgrep shouldafound</code></a>.</li>
+        <li>A bug within semgrep itself.</li>
```
```suggestion
        <li>A bug within Semgrep itself.</li>
```

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -12,25 +12,49 @@ import MoreHelp from "/src/components/MoreHelp"
+
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
+* **Triage the finding.** If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include ignoring and reopening. Triaging a finding to ignore it is one method to handle **false positives** without changing the rule or code.
```
I love the active and direct-action sentences that start each bullet here!

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -39,48 +63,54 @@ The Semgrep App Finding page displays findings across all projects connected to
+| **Severity**    | The impact of a finding. Possible values: <ul><li>Low</li><li>Medium</li><li>High </li><li>Critical</li></ul> |
+| **Finding**     | The file, line, and branch of the match in the code. Clicking this field brings you to the exact location in the codebase. |
+| **Project**     | The name of the repository where the finding was found. Clicking this field brings you to the project’s URL. |
+| **Rule**        | The name of the rule that  matched with the code. Clicking this field brings you to the rule’s entry in the [Rule Registry](https://semgrep.dev/r). |
```
Regarding the link to Registry, what about for rules that aren’t in the Registry, like private rules or rules hosted somewhere else?

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -39,48 +63,54 @@ The Semgrep App Finding page displays findings across all projects connected to
+| **Project**     | The name of the repository where the finding was found. Clicking this field brings you to the project’s URL. |
+| **Rule**        | The name of the rule that  matched with the code. Clicking this field brings you to the rule’s entry in the [Rule Registry](https://semgrep.dev/r). |
+| **Introduced**  | The time when this finding was first found. Clicking the header row sorts the table by latest or earliest findings. |
+| **Status**      | Refers to a finding’s triage state. Refer to the following table to understand triage states. |
```
You could link to the “following table” since in a future docs edit, these sections may not be so close to each other.

s-santillan replied:
Will do!

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -39,48 +63,54 @@ The Semgrep App Finding page displays findings across all projects connected to
+| **Ignored** | Findings that are ignored will not be acted upon. This may be a false positive or deprioritized issue. Findings can be ignored through Semgrep App (see [Managing triage states](#managing-triage-states-bulk-triage)). There are two types of ignored findings: <dl><dt>Ignored - App</dt><dd>The ignore status was set within Semgrep App.</dd><dt>Ignored - Code</dt><dd>The ignore status w ...
+| **Fixed** | Fixed findings are findings that were detected in a previous scan but no longer trigger a match in the most recent scan. The rule that detected the finding and the code that triggered the match must both be active in the most recent scan. |
+
+Findings can also be **removed**. A removed finding does not count towards a fix rate or the total number of findings. A finding is considered removed if it is not found in the most recent scan due to either of the following conditions:
```
Could it also be that the matching code (but not the entire file containing it) was removed?

s-santillan replied:
(See the following comment. I am resolving this as I addressed the concern in the note below.)

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -39,48 +63,54 @@ The Semgrep App Finding page displays findings across all projects connected to
+
+* You removed the rule from the Rule Board.
+* You deleted the file containing the code.
+* The file is included in a `.semgrepignore` file.
```
Could it also be that a //semgrepignore comment was added to (above?) the matching line?

s-santillan replied:
(I assume you mean the comment annotation `nosemgrep`.) 

There is actually a difference in ignoring via comment (`// nosemgrep`) and ignoring via file. //nosemgrep is considered a triage action, so Semgrep still scans the code, makes the finding record, and ignores it. It appears as "Ignored - Code" in the findings page. For that reason, I don't include adding a 'nosemgrep' comment to the code.

I did bring up before (a long time ago) that there's one word (ignore) for two different things. Maybe "muted" is a term we'd like to reexplore, but right now, this is how the product works.

pabloest replied:
Yes, I meant `// nosemgrep` :)

s-santillan replied:
(Resolving since for now //nosemgrep does not do the same thing as adding a file in .semgrepignore :) )

pabloest commented on docs/semgrep-app/findings.md:
```diff
@@ -119,36 +149,49 @@ To **open findings**:
+    <dd>Diff-aware scans are performed on non-mainline branches, such as in pull requests and merge requests. Diff scans traverse the repository's files based on the commit where the branch diverged from the mainline branch (or diverged from the last commit that was fully scanned?)</dd>
+</dl>
+
+If a finding is fixed in one branch (such as `main`) but open in another (such as `production`), and the code fixes are present in both branches, initiate a scan on the branch(es) with open findings to mark the findings as fixed.
```
Would Semgrep mark these open findings as now fixed, or does the user have to mark them? I’m not quite sure from the last phrase.

s-santillan replied:
Semgrep will mark the findings as fixed. I will add this in.

adamkvitek commented on docs/semgrep-app/findings.md:
```diff
@@ -10,27 +10,53 @@ import MoreHelp from "/src/components/MoreHelp"
+* **Fix the issue detected by the finding.** This is Semgrep's primary goal. In this case, the rule produces a **true positive** finding (such as a security issue) as intended and developers must refactor or address the code such that the rule no longer matches it.
+* **View the Semgrep rule and the matching code.** For developers aiming to understand their team's security posture, Semgrep provides a top-level report view through the Dashboard and a list view of findings in the Findings page that can be filtered by repository, rule, branch, or triage action.
+* **Triage the finding.** If the finding is not useful or important, it can be deprioritized through triaging. Triage actions include ignoring and reopening. Triaging a finding to ignore it is one method to handle **false positives** without changing the rule or code.
+* **Remove the rule or code that generated the finding**. There are cases where Semgrep scans a file not meant for scanning or when a rule is irrelevant. You can remove the rule from the Rule board or add the file to the ignore list. To remove a rule, see [Managing triage states](../findings/#managing-triage-states-bulk-triage).
```
```suggestion
* **Remove the rule or code that generated the finding.** There are cases where Semgrep scans a file not meant for scanning or when a rule is irrelevant. You can remove the rule from the Rule board or add the file to the ignore list. To remove a rule, see [Managing triage states](../findings/#managing-triage-states-bulk-triage).
```
